File: PerfTest1/PerfTest1-Xamarin/PerfTest1-ClassicXamarin.Android/MainTabListener.cs
using Android.App;

namespace PerfTest1Xamarin
{
    public class MainTabListener : Java.Lang.Object, ActionBar.ITabListener
    {
        private Fragment currentFragment;

        public MainTabListener()
        {
        }

        public void OnTabSelected(ActionBar.Tab tab, FragmentTransaction ft)
        {
            currentFragment = null;
            if (tab.Position == 0)
            {
                currentFragment = new AzureList();
                ft.Replace(Resource.Id.main_area, currentFragment);
            }
            else if (tab.Position == 1)
            {
                currentFragment = new CalcPrimes();
                ft.Replace(Resource.Id.main_area, currentFragment);
            }
        }

        public void OnTabReselected(ActionBar.Tab tab, FragmentTransaction ft)
        {
        }

        public void OnTabUnselected(ActionBar.Tab tab, FragmentTransaction ft)
        {
            ft.Remove(currentFragment);
        }
    }
}

File: PerfTest1/PerfTest1-Java/app/src/main/java/com/vandammeford/kevinf/perftest1_java/MainTabListener.java
package com.vandammeford.kevinf.perftest1_java;

import android.app.ActionBar;
import android.app.Fragment;
import android.app.FragmentTransaction;
import android.widget.Button;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by KevinF on 12/12/2014.
 */
public class MainTabListener implements ActionBar.TabListener {
    Fragment currentFragment;

    public MainTabListener() {
    }

    @Override
    public void onTabSelected(ActionBar.Tab tab, FragmentTransaction fragmentTransaction) {
        currentFragment = null;
        if (tab.getPosition() == 0) {
            currentFragment = new AzureList();
            fragmentTransaction.replace(R.id.main_area, currentFragment);
        } else if (tab.getPosition() == 1){
            currentFragment = new CalcPrimes();
            fragmentTransaction.replace(R.id.main_area, currentFragment);
        }
    }

    @Override
    public void onTabUnselected(ActionBar.Tab tab, FragmentTransaction fragmentTransaction) {
        fragmentTransaction.remove(currentFragment);
    }

    @Override
    public void onTabReselected(ActionBar.Tab tab, FragmentTransaction fragmentTransaction) {

    }
}